hazelcast/src/main/java/com/hazelcast/internal/dynamicconfig/ClusterWideConfigurationService.java

```diff
@@ -0,0 +1,531 @@
+    private final ConcurrentMap<String, ReplicatedMapConfig> replicatedMapConfigs =
+            new ConcurrentHashMap<String, ReplicatedMapConfig>();
+    private final ConcurrentMap<String, TopicConfig> topicConfigs = new ConcurrentHashMap<String, TopicConfig>();
+    private final ConcurrentMap<String, ExecutorConfig> executorConfigs = new ConcurrentHashMap<String, ExecutorConfig>();
+    private final ConcurrentMap<String, DurableExecutorConfig> durableExecutorConfigs =
+            new ConcurrentHashMap<String, DurableExecutorConfig>();
+    private final ConcurrentMap<String, ScheduledExecutorConfig> scheduledExecutorConfigs =
+            new ConcurrentHashMap<String, ScheduledExecutorConfig>();
+    private final ConcurrentMap<String, SemaphoreConfig> semaphoreConfigs = new ConcurrentHashMap<String, SemaphoreConfig>();
+    private final ConcurrentMap<String, QueueConfig> queueConfigs = new ConcurrentHashMap<String, QueueConfig>();
+    private final ConcurrentMap<String, ReliableTopicConfig> reliableTopicConfigs =
+            new ConcurrentHashMap<String, ReliableTopicConfig>();
+    private final ConcurrentMap<String, CacheSimpleConfig> cacheSimpleConfigs =
+            new ConcurrentHashMap<String, CacheSimpleConfig>();
+    private final ConcurrentMap<String, EventJournalConfig> cacheEventJournalConfigs =
+            new ConcurrentHashMap<String, EventJournalConfig>();
+    private final ConcurrentMap<String, EventJournalConfig> mapEventJournalConfigs =
+            new ConcurrentHashMap<String, EventJournalConfig>();
+
+    private final ConfigPatternMatcher configPatternMatcher;
+
+    @SuppressWarnings("unchecked")
+    private final Map<?, ? extends IdentifiedDataSerializable>[] allConfigurations = new Map[]{
+            mapConfigs,
+            multiMapConfigs,
+            cardinalityEstimatorConfigs,
+            ringbufferConfigs,
+            lockConfigs,
+            listConfigs,
+            setConfigs,
+            replicatedMapConfigs,
+            topicConfigs,
+            executorConfigs,
+            durableExecutorConfigs,
+            scheduledExecutorConfigs,
+            semaphoreConfigs,
+            queueConfigs,
+            reliableTopicConfigs,
+            cacheSimpleConfigs,
+            cacheEventJournalConfigs,
+            mapEventJournalConfigs,
+    };
+
+    private volatile Version version;
+
+    public ClusterWideConfigurationService(NodeEngine nodeEngine, DynamicConfigListener dynamicConfigListener) {
+        this.nodeEngine = nodeEngine;
+        this.listener = dynamicConfigListener;
+        this.configPatternMatcher = nodeEngine.getConfig().getConfigPatternMatcher();
+    }
+
+    @Override
+    public Operation prepareReplicationOperation(PartitionReplicationEvent event) {
+        if (version.isLessOrEqual(V3_8)) {
+            return null;
+        }
+        IdentifiedDataSerializable[] allConfigurations = getAllConfigurations();
+        if (allConfigurations.length == 0) {
```

Comment:
it can happen when there is no dynamic config registered. it's an optimization so we do not send an empty operation.